src/function/scalar/string/string_split.cpp

```diff
@@ -0,0 +1,288 @@
+		result.chunks.push_back(move(append_chunk));
+	}
+	result.Verify();
+}
+
+unique_ptr<ChunkCollection> string_split(string_t input, string_t delim, const bool regex) {
+	const char *input_data = input.GetData();
+	size_t input_size = input.GetSize();
+	const char *delim_data = delim.GetData();
+	size_t delim_size = delim.GetSize();
+
+	bool ascii_only = true;
+	for (auto i = 0; i < (int)input_size; ++i) {
+		if (input_data[i] & 0x80) {
+			ascii_only = false;
+			break;
+		}
+	}
+
+	auto output = make_unique<ChunkCollection>();
+	vector<LogicalType> types = {LogicalType::VARCHAR};
+	output->types = types;
+
+	Iterator *iter;
+	if (regex) {
+		auto re = make_unique<RE2>(duckdb_re2::StringPiece(delim_data, delim_size));
+		if (!re->ok()) {
+			throw Exception(re->error());
+		}
+		iter = new RegexIterator(input_size, move(re), ascii_only);
+	} else if (ascii_only) {
+		iter = new AsciiIterator(input_size, delim_data, delim_size);
+	} else {
+		iter = new UnicodeIterator(input_size, delim_data, delim_size);
+	}
+	string_split(input_data, *iter, *output);
+	delete iter;
+
+	return output;
+}
+
+static void string_split_executor(DataChunk &args, ExpressionState &state, Vector &result, const bool regex) {
+	VectorData input_data;
+	args.data[0].Orrify(args.size(), input_data);
+	auto inputs = (string_t *)input_data.data;
+
+	VectorData delim_data;
+	args.data[1].Orrify(args.size(), delim_data);
+	auto delims = (string_t *)delim_data.data;
+
+	result.Initialize(LogicalType::LIST);
+	auto list_struct_data = FlatVector::GetData<list_entry_t>(result);
+
+	auto list_child = make_unique<ChunkCollection>();
+	vector<LogicalType> types = {LogicalType::VARCHAR};
+	list_child->types = types;
+
+	size_t total_len = 0;
+	for (idx_t i = 0; i < args.size(); i++) {
+		if ((*input_data.nullmask)[input_data.sel->get_index(i)]) {
+			FlatVector::SetNull(result, i, true);
+			continue;
+		}
+		string_t input = inputs[input_data.sel->get_index(i)];
+
+		unique_ptr<ChunkCollection> split_input;
+		if ((*delim_data.nullmask)[delim_data.sel->get_index(i)]) {
+			// special case: delimiter is NULL
+			split_input = make_unique<ChunkCollection>();
+			split_input->types = types;
+
+			auto append_chunk = make_unique<DataChunk>();
+			append_chunk->Initialize(types);
+			FlatVector::GetData<string_t>(append_chunk->data[0])[append_chunk->size()] =
+		    	StringVector::AddString(append_chunk->data[0], input);
+			append_chunk->SetCardinality(append_chunk->size() + 1);
+
+			split_input->count += append_chunk->size();
+			split_input->chunks.push_back(move(append_chunk));
+		} else {
+			string_t delim = delims[delim_data.sel->get_index(i)];
+			split_input = string_split(input, delim, regex);
+		}
+		list_struct_data[i].length = split_input->count;
+		list_struct_data[i].offset = total_len;
+		total_len += split_input->count;
+		list_child->Append(*split_input);
+	}
+
+	assert(list_child->count == total_len);
+	result.vector_type = VectorType::CONSTANT_VECTOR;
```

Comment:
The result should only be a constant vector if both inputs are constants, no?

Reply:
You are right. Changed to check this before setting